### PR TITLE
Update dart sdk version to 2.0.0-dev.23.0

### PR DESCRIPTION
**Pull request by a-siva**

### CODE REVIEW

**Comment by MichaelRFairhurst** [comment]
Regarding the void changes -- I thought I had fixed all cases of this already: https://github.com/flutter/flutter/pull/14366/files

I will make another PR to tackle these issues separately, you can either cherry-pick my changes or rebase once that PR has landed to get rid of these errors.

**Comment by MichaelRFairhurst** [comment]
It looks like this will also require updating mockito.

File: packages/flutter/test/material/animated_icons_private_test.dart [hunk 1]
```diff
@@ -104,7 +104,7 @@ void main () {
       painter.paint(mockCanvas, size);
       expect(generatedPaths.length, 1);
 
-      verifyInOrder(<dynamic>[
+      verifyInOrder(<void>[
```
Comment (leafpetersen) [inline packages/flutter/test/material/animated_icons_private_test.dart]: I think you will need to pass a type argument to verifyInOrder to satisfy their lints.

```
verifyInOrder<void>(<void>[...
```

**Comment by tvolkert** [comment]
See also https://github.com/flutter/flutter/pull/14610, which moves to using the Dart SDK that's built into the engine as the host SDK used in flutter_tools.

/cc @aam

**Comment by a-siva** [comment]
abandoning this CL as we use the Dart SDK from the engine.

**Comment by leafpetersen** [comment]
Does this mean we are not rolling the framework to dev.23?  What are the roll plans going forward?

**Comment by tvolkert** [comment]
It means that the host SDK (used by flutter_tools) is now pinned to the Dart version that's built into the engine (which is ahead of dev.23 and is generally pretty close to tip-of-tree).  Going forward, it'll remain pinned to the engine Dart SDK version, which should remain fairly close to tip-of-tree.

**Pull request closed without merging**